### PR TITLE
puller(ticdc): fix renameTables ddl job

### LINKED ISSUE #9476: [CDC:ErrSnapshotSchemaNotFound] schema 41766 not found in schema snapshot
### What did you do?

1. create changefeed with the following config:
```toml
[filter]
rules = ['*.*', '!ignore1.*']
```
2. create database ignore1; create table a and a_new;
3. USE ignore1; RENAME TABLE a TO a_bak, a_new TO a;

### What did you expect to see?

cdc skip the rename tables ddl

### What did you see instead?

[CDC:ErrSnapshotSchemaNotFound] schema 41766 not found in schema snapshot

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
```

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #9476

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: cdc/puller/ddl_puller.go [hunk 1]
```diff
@@ -270,37 +270,40 @@ func (p *ddlJobPullerImpl) handleRenameTables(job *timodel.Job) (skip bool, err
 	remainTables := make([]*timodel.TableInfo, 0, len(multiTableInfos))
 	snap := p.schemaStorage.GetLastSnapshot()
 	for i, tableInfo := range multiTableInfos {
-		schema, ok := snap.SchemaByID(newSchemaIDs[i])
+		var shouldDiscardNewTable, shouldDiscardOldTable bool
+		newSchemaName, ok := snap.SchemaByID(newSchemaIDs[i])
 		if !ok {
-			return true, cerror.ErrSnapshotSchemaNotFound.GenWithStackByArgs(newSchemaIDs[i])
```
Comment (asddongmen) [inline cdc/puller/ddl_puller.go:275]: // can't find the new table schema, means the new table name does not hit the filter rule and we should discard the new table

**Review by asddongmen — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**